src/QtHtmlReader.cpp

```diff
@@ -0,0 +1,267 @@
+QtHtmlReader::QtHtmlReader() : width(1024), height(768), x_offset(0), y_offset(0), html(""), css(""), background_color("#000000"), is_open(false), gravity(GRAVITY_CENTER)
+{
+	// Open and Close the reader, to populate it's attributes (such as height, width, etc...)
+	Open();
+	Close();
+}
+
+QtHtmlReader::QtHtmlReader(int width, int height, int x_offset, int y_offset, GravityType gravity, std::string html, std::string css, std::string background_color)
+: width(width), height(height), x_offset(x_offset), y_offset(y_offset), gravity(gravity), html(html), css(css), background_color(background_color), is_open(false)
+{
+	// Open and Close the reader, to populate it's attributes (such as height, width, etc...)
+	Open();
+	Close();
+}
+
+// Open reader
+void QtHtmlReader::Open()
+{
+	// Open reader if not already open
+	if (!is_open)
+	{
+		// create image
+		image = std::shared_ptr<QImage>(new QImage(width, height, QImage::Format_RGBA8888));
+		image->fill(QColor(background_color.c_str()));
+
+		//start painting
+		QPainter painter;
+		if (!painter.begin(image.get())) {
+			return;
+		}
+
+		//set background
+		painter.setBackground(QBrush(background_color.c_str()));
+
+		//draw text
+		QTextDocument text_document;
+
+		//disable redo/undo stack as not needed
+		text_document.setUndoRedoEnabled(false);
+
+		//create the HTML/CSS document
+		text_document.setTextWidth(width);
+		text_document.setDefaultStyleSheet(css.c_str());
+		text_document.setHtml(html.c_str());
+
+		int td_height = text_document.documentLayout()->documentSize().height();
+
+ 		if (gravity == GRAVITY_TOP_LEFT || gravity == GRAVITY_TOP || gravity == GRAVITY_TOP_RIGHT) {
+ 			painter.translate(x_offset, y_offset);
+ 		} else if (gravity == GRAVITY_LEFT || gravity == GRAVITY_CENTER || gravity == GRAVITY_RIGHT) {
+ 			painter.translate(x_offset, (height - td_height) / 2 + y_offset);
+ 		} else if (gravity == GRAVITY_BOTTOM_LEFT || gravity == GRAVITY_BOTTOM_RIGHT || gravity == GRAVITY_BOTTOM) {
+ 			painter.translate(x_offset, height - td_height + y_offset);
+ 		}
+
+ 		if (gravity == GRAVITY_TOP_LEFT || gravity == GRAVITY_LEFT || gravity == GRAVITY_BOTTOM_LEFT) {
+ 			text_document.setDefaultTextOption(QTextOption(Qt::AlignLeft));
+ 		} else if (gravity == GRAVITY_CENTER || gravity == GRAVITY_TOP || gravity == GRAVITY_BOTTOM) {
+ 			text_document.setDefaultTextOption(QTextOption(Qt::AlignHCenter));
+ 		} else if (gravity == GRAVITY_TOP_RIGHT || gravity == GRAVITY_RIGHT|| gravity == GRAVITY_BOTTOM_RIGHT) {
+ 			text_document.setDefaultTextOption(QTextOption(Qt::AlignRight));
+ 		}
+
+ 		// Draw image
+		text_document.drawContents(&painter);
+
+		painter.end();
+
+		// Update image properties
+		info.has_audio = false;
+		info.has_video = true;
+		info.file_size = 0;
+		info.vcodec = "QImage";
+		info.width = width;
+		info.height = height;
+		info.pixel_ratio.num = 1;
+		info.pixel_ratio.den = 1;
+		info.duration = 60 * 60 * 24; // 24 hour duration
```

Comment:
```suggestion
		info.duration = 60 * 60 * 1;  // 1 hour duration
```

Reply:
(To match the band-aid I added to #346.)

Reply:
:+1: